pkg/oc/bootstrap/clusteradd/componentinstall/apply_template.go

```diff
@@ -25,15 +26,24 @@ type Template struct {
 	RBACTemplate      []byte
 	InstallTemplate   []byte
 
+	ComponentImage string
+
 	WaitCondition func() (bool, error)
 }
 
-func (t Template) MakeReady(image, baseDir string, params map[string]string) Component {
+func (t Template) MakeReady(imageFormat, baseDir string, params map[string]string) Component {
+	paramsWithImage := make(map[string]string)
+	for k, v := range params {
+		paramsWithImage[k] = v
+	}
+	if len(t.ComponentImage) > 0 {
+		paramsWithImage["IMAGE"] = strings.Replace(imageFormat, "${component}", t.ComponentImage, -1)
```

Comment:
Non-goal.  If you want to specify an image in your params, add it to your params.  There's no need for this special case.

Reply:
Each component is different.  That means we have to loop over the components and get info from that component to find the image.